gen.sh

```diff
@@ -63,7 +64,7 @@ _get_ext () {
 
 _base () {
   ext=$(_get_ext "$1")
-  echo $(basename "$1" .$ext)
+  basename "$1" ."$ext"
```

Comment:
```suggestion
  basename "$1" ".$ext"
```
and similar elsewhere

Reply:
I fixed all that I noticed, if you saw any other places please point them out.